templates/deployment.yaml

```
@@ -1,13 +1,18 @@
 {{- range $serviceName, $serviceDefinition := .Values.services }}
 {{- if eq $serviceDefinition.enabled true }}
 ---
+{{- if eq ( $serviceDefinition.type | default "Deployment" ) "Job" }}
+apiVersion: batch/v1
+{{- else }}
 apiVersion: apps/v1
-kind: {{ ( default $serviceDefinition.type "Deployment" ) }}
+{{- end }}
+
+kind: {{ $serviceDefinition.type | default "Deployment" }}
 metadata:
   name: {{ include "common.fullname" ( dict "root" $ "service" $serviceDefinition "serviceName" $serviceName ) }}
   {{- include "common.metadata" ( dict "root" $ "service" $serviceDefinition "serviceName" $serviceName ) | nindent 2 }}
 spec:
-  {{- if eq ( default $serviceDefinition.type "Deployment" ) "Deployment" }}
+  {{- if eq ( $serviceDefinition.type | default "Deployment" ) "Deployment" }}
```

This mix it difficult to read => 
```suggestion
  {{- if eq ( default "Deployment" $serviceDefinition.type ) "Deployment" }}
```

---

yep